### ISSUE TITLE
Add push rules support

### ISSUE THREAD

**Issue by kontez** [python-gitlab/python-gitlab#397]
I think that these endpoints are not yet supported by the python wrapper, it would be a nice to have :)

[https://docs.gitlab.com/ee/api/projects.html#get-project-push-rules](https://docs.gitlab.com/ee/api/projects.html#get-project-push-rules)

**Comment by ani-patel** [comment]
project = glab.projects.get(id)
body = configuration['push_rules'] // JSON body for push rule. You can check dcoumentation for API for this.
# Update the push rule with the configuration through Gitlab API.
r = requests.put(GITLAB_BASE_URL+'/api/v4/projects/'+str(project.id)+'/push_rule', data = body, headers = { 'PRIVATE-TOKEN': GITLAB_API_KEY })
if r.status_code == 200:
    print 'Push Rule added'
else:
    print 'Update push rule failed'

**Comment by kontez** [comment]
great, thanks!

**Comment by felixhummel** [comment]
Another workaround (using python-gitlab's instance `gl`):
```python
def _push_rule_url(project_id):
    return '/projects/{0}/push_rule'.format(project_id)

def get(project_id):
    return gl.http_get(_push_rule_url(project_id))

def put(project_id):
    return gl.http_put(_push_rule_url(project_id), push_rule)

if __name__ == '__main__':
    project = gl.projects.get('fooproject')
    push_rule = get(project.id)
    print(push_rule)

```

**Comment by gpocentek** [comment]
Fixed with #520
